### PR TITLE
Close dependency-detection gaps + quarto vignettes + inside_rmd autodetect

### LINKED ISSUE #106: As user, I don't want to have to think if I need `inside_rmd` or not when using `att_from_rmd()`
As user, I don't want to have to think if I need `inside_rmd` or not when using `att_from_rmd()`, so that the same function can be presented in help as in the classical use.

## Criteria of success

- [ ] There is no more `inside_rmd` parameter to be chosen by the user
- [ ] `att_from_rmd()` works the same, inside or outside rmd
- [ ] Readme and doc of {attachment} is simplified to deal with this change
- [ ] Make {callr} a Suggest of the package

## Tech

- [ ] use `outside_knit <- is.null(knitr::opts_knit$get("out.format"))` to detect if we are in or outside a knitting environment, thus running directly the code, or run using `callr:r()` call. 
- [ ] {callr} is only needed if using the function when running knit, which is not really frequent with {attachment}, except for documentation and courses.

### LINKED ISSUE #120: att_from_* functions shouldn't identify packages within quotes and comments
When using any `att_from_*` function on a script containing double colons, it identifies packages that aren't packages.
This subsequently breaks any combination with `att_to_desc_from_is` or similar utility function.

Examples with an xpath using xml2 library :
```R
xml_find_all(ttt, "//tr/th[@id='r6']/following-sibling::td") #att_from_rmd will return `sibling` as a package...
```

Example with a comment : 
```R
filter(df, value > 5) #this is not base::Filter ! <- will return base as a package
```

I believe the cause of this unwanted behaviour is the regex in [`att_from_rscripts.R`](https://github.com/ThinkR-open/attachment/blob/main/R/att_from_rscripts.R) :
`str_extract_all("[[:alnum:]\\.]+(?=::)") %>%`

The regex doesn't seem check the context around supposed "package" as you can see [this playground](https://regex101.com/r/T44SUt/1)
It should be possible to either build a cryptic regex checking if you are within parentheses or comment, or remove comments and quotes with gsub beforehand.
In either cases this might lose support for libraries called within string interpolation such as `glue("Value : {data.table::fifelse(myvar, 1,2)}`

### LINKED ISSUE #128: Also detect instances of `use()`
Currently, `attachment::att_from_rscript()` doesn't discover instances of `use()` introduced with R 4.4.0:

```r
use("tidyr", c("pivot_longer", "pivot_wider"))
```

### LINKED ISSUE #131: `att_amend_desc()` incorrectly forces rmarkdown in Suggests for Quarto vignettes
When using `attachment::att_amend_desc()` with a Quarto vignette (.qmd), the function ignores the file extension, and automatically adds `{rmarkdown}` and removes `{quarto}` to/from Suggests, even when the vignette clearly uses Quarto.

```
> attachment::att_amend_desc()
Documentation parameters were restored from attachment config file.
Updating test.att.quarto documentation
ℹ Loading test.att.quarto
ℹ Loading test.att.quarto
[-] 1 package(s) removed: quarto.
[+] 1 package(s) added: rmarkdown.
```

Fortunately, there is a workaround via the config file:
```
extra.suggests: quarto
pkg_ignore: rmarkdown
```

Should it be possible for `att_amend_desc()` to detect .qmd vignettes and avoid adding `{rmarkdown}` and removing `{quarto}`  in such cases?

Thanks for considering this issue, and for providing this function, which is really helpful!

### LINKED ISSUE #132: att_amend_desc seems to detect dependencies from css rules ?
Image

### PR DESCRIPTION
**Pull request by VincentGuyader**
## Summary

Broad cleanup + bug-fix branch. Modernises `{attachment}`'s dependency detection (AST-based), fixes Quarto vignette handling, simplifies `inside_rmd` via auto-detect, and drops the `{fusen}` development workflow.

## Closed issues

Closes #106 — `inside_rmd` auto-detect via `knitr::opts_knit$get("out.format")`; tested in `test-inside-rmd-autodetect.R`.

Closes #120 — `::` inside string literals / comments (xpath `following-sibling::`, CSS `::after`, `sprintf('%s::plot()', ...)`, etc.) is no longer misread as a package reference; tested in `test-edge-cases.R` + `f_edge_cases.R`.

Closes #128 — `use("pkg", ...)` (R ≥ 4.4) is now detected; tested in `f_edge_cases.R` (findedge8–10).

Closes #131 — `.qmd` vignette directories add `quarto` instead of / alongside `rmarkdown`; tested in `test-quarto-vignettes.R`.

Closes #132 — CSS-style `::` patterns inside strings are ignored; tested in `f_edge_cases.R` (`label::after`, `::first-line`).

## Partially addressed (not closed)

- #129 — the false positives it lists (`s` from sprintf, `matplotlib` from string literals, `use()`, `getFromNamespace()`, named-arg `library(package = ...)`) are **fixed and tested**. Two open points are **out of scope** for this PR:
  - raising an error when a declared package is not installed — belongs to `att_amend_desc()` via `check_if_suggests_is_installed`, not the pure parser;
  - resolving bare function names to their origin package (`toTitleCase → tools`) — requires a function→package lookup layer and changes the whole detection paradigm.
- #133 — the CSS `::` issue it reports is subsumed by the AST refactor, and the PR's regex patch is redundant once this branch merges. Leaving it to the maintainer to decide whether to close it.

## What's in the branch

- **AST-based dependency detection** (`att_from_rscript()`): walks the R parse tree, with a regex-based legacy fallback (now accepting `_` in package names) when `parse()` fails. Adds detection of `use("pkg")` (R ≥ 4.4), `getFromNamespace(fn, "pkg")`, `loadNamespace()`, named-arg forms (`library(package = "pkg")`, `requireNamespace("pkg", lib.loc = "...")`), and fully-qualified intro calls (`base::library(pkg)`, `methods::getFromNamespace(...)`). Introspection helpers (`packageVersion`, `getNamespace`, `asNamespace`, `attachNamespace`) are intentionally **not** treated as dep introducers. Gains an `encoding` argument (defaults to `getOption("encoding")`) so Latin-1 / Windows-1252 scripts on Windows aren't forced through UTF-8. Emits a `warning()` when the AST parser fails and we fall back to legacy regex.
- **Walker robustness**: handles "empty" AST nodes (`x[, 1]`, missing `else`, `switch(a = , b = ...)`) via a `walk_elem()` helper. Without this fix the walker crashed with `argument "el" is missing, with no default` on any realistic script.
- **Quarto-aware vignettes** (`att_from_rmds()`): engine inferred from files actually present.
- **`inside_rmd` auto-detect** (`att_from_rmd()` / `att_from_rmds()`): default `NULL`, detected via `knitr::opts_knit$get("out.format")`.
- **Drop `{fusen}` workflow**: `fusen::sepuku()` applied; `dev/flat_*.Rmd` and `dev/config_fusen.yaml` removed; banners stripped from generated files (now hand-maintained).
- **Manual detection harness**: `dev/manual_detection_edge_cases.R` — a ~600-line R script exercising every edge case we could think of (parse-tree weirdness, string literals, raw strings, comments, empty args, quote/bquote, deep/wide nesting, fully-qualified intro calls, etc.), with the expected true- and false-positive lists baked in as comments.

## Tests

`[ FAIL 0 | SKIP 1 | PASS 389 ]` on clean CI. Every fix has at least one test; the detection harness checks 125 true positives and ~45 false positives and currently scores 100%.

## Version

Version bumped to `1.0.0` to signal the detector's new parse-tree foundation.

## Test plan

- [x] `devtools::test()` green on CI across macOS, Windows, ubuntu devel/release/oldrel-{1,2,3}
- [x] R CMD check green on all platforms
- [x] Manual: `att_amend_desc()` on a package with a `.qmd` vignette (should no longer force `rmarkdown` / remove `quarto`)
- [x] Manual: `att_from_rscript()` on scripts with xpath-style `::` inside strings (should not flag fake packages)
- [x] Manual: `att_from_rscript()` on scripts with `x[, 1]` / empty `switch` arms (should not crash)

### CODE REVIEW

File: R/att_from_rscripts.R [hunk 1]
```diff
@@ -18,27 +21,114 @@
 
 att_from_rscript <- function(path) {
 
-  file <- as.character(parse(path))
-  # Replace newlines `\n` by space
-  file <- gsub("\\\\n", " ", file)
+  lines <- tryCatch(
+    readLines(path, warn = FALSE, encoding = "UTF-8"),
+    error = function(e) character(0)
```
[the remark this answers is not in the document]
Comment (VincentGuyader) [reply R/att_from_rscripts.R]: @copilot je ne veux pas d'erreur, mais oui ça peut être bien de lister les soucis de lecture, sans les rendre bloquantes. propose moi quelque chose d'autre dans ce sens
